safehttp/plugins/reportingapi/reporting.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package reportingapi is an implementation of the Report-To header described
+// in https://www.w3.org/TR/reporting/#header.
+//
+// It allows for setting reporting groups to use in conjuction with COOP and CSP.
+package reportingapi
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/go-safeweb/safehttp"
+)
+
+// DefaultMaxAge is used as default cache duration for report groups and will make them last 7 days.
+const DefaultMaxAge = 7 * 24 * 60 * 60
+
+const ReportToHeaderKey = "Report-To"
+
+// Endpoint is the Go representation of the endpoints values as specified
+// in https://www.w3.org/TR/reporting/#endpoints-member
+type Endpoint struct {
+	// URL defines the location of the endpoint.
+	// The URL that the member’s value represents MUST be potentially trustworthy. Non-secure endpoints will be ignored.
```

Comment:
No, this is spec language.

The URL MUST be pointing to something defined as "potentially trustworthy". This means that for example it has to start with "https" because "http" endpoints can't be potentially trustworthy.

Note that not all https urls are trustworthy, but all of them are potentially so.

Reply:
I see. Confusing, but if it's the spec language then that's fine.